packages/sanity/src/core/releases/components/Table/Table.tsx

```diff
@@ -58,7 +60,14 @@ const TableInner = <TableData, AdditionalRowTableData>({
   rowActions,
   loading = false,
 }: TableProps<TableData, AdditionalRowTableData>) => {
-  const {searchTerm, sort} = useTableContext()
+  const {searchTerm, sort, setDefaultSort} = useTableContext()
+
+  const {column: defaultSortColumn, direction: defaultSortDirection} = defaultSort || {}
+  useEffect(() => {
```

Comment:
Small thought here: is there a good reason to have this effect that chains one piece of state to another? Could you group the functions that set the state together using a function or a reducer?

I haven't read the code _super_ closely, so apologies if I've missed something obvious.

Reply:
And/or - would it be better if it was passed directly to TableProvider and applied as a default there?